walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,80 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays; 
+
 
 public final class FindMeetingQuery {
+
+  /**
+   * Returns {@code TimeRange} when a meeting can happen based on other events and the
+   * meeting participants involved.
+   * @param events {@code Collection} of events happening.
+   * @param request {@code MeetingRequest} for a meeting
+   * @return Collection of {@code TimeRange} when a meeting can happen. 
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    final int MINUTES_IN_A_DAY = 1440;
+    ArrayList<TimeRange> possibleTimes = new ArrayList<TimeRange>();
+    ArrayList<Event> localEvents = new ArrayList<Event>(events);
+    boolean[] availableTimes = new boolean[MINUTES_IN_A_DAY+1];
+
+    //Meeting is longer than a day, that can't happen
+    if(request.getDuration() > MINUTES_IN_A_DAY){
+      return possibleTimes;
+    }
+
+    //No atendees on the request
+    if(request.getAttendees().size() == 0 && request.getOptionalAttendees().size() == 0){
+      possibleTimes.add(TimeRange.WHOLE_DAY);
+      return possibleTimes;
+    }
+
+    //Clean schedule, all day is free
+    if(localEvents.isEmpty()){
+      possibleTimes.add(TimeRange.WHOLE_DAY);
+      return possibleTimes;
+    }
+
+    for(Event event : events){
+      //If event has no atendees involved in the request.
+      if(!Collections.disjoint(event.getAttendees(), request.getAttendees())){
+        Arrays.fill(availableTimes, event.getWhen().start(), event.getWhen().end(), true);
+      }else{
+        localEvents.remove(event);
+      }
+    }
+
+    //Find free time in the array and if they fit the meeting, create the time range
+    int intervalDuration = 0;
+    for(int i = 0; i < availableTimes.length; i++){
+      if(!availableTimes[i]){
```

Comment:
I'm not sure I understand the goal here.

Because of the `!availableTimes[i]` check, it seems like `intervalDuration` is supposed to represent a period of **unavailable** times, but then these are added to the `possibleTimes` array?

Reply:
availableTimes holds true where there's a meeting. Wrong choice of name, which has been fixed.